rust/common-wasi/src/bindings.rs

```diff
@@ -0,0 +1,167 @@
+/// Synchronous-generated bindings for WASI interfaces.
+pub mod sync {
+    mod generated {
+        use crate::StreamError;
+
+        wasmtime::component::bindgen!({
+            path: "wit",
+            world: "common:wasi/imports",
+            tracing: true,
+            trappable_error_type: {
+                "wasi:io/streams/stream-error" => StreamError,
+                "wasi:filesystem/types/error-code" => crate::stubs::filesystem::FsError,
+                "wasi:sockets/network/error-code" => crate::stubs::sockets::SocketError,
+                "wasi:http/types/error-code" => crate::stubs::http::HttpError,
+            },
+            trappable_imports: true,
+            with: {
+                // These interfaces come from the outer module, as it's
+                // sync/async agnostic.
+                "wasi:clocks": crate::bindings::clocks,
+                "wasi:random": crate::bindings::random,
+                "wasi:cli": crate::bindings::cli,
+                "wasi:io/error": crate::bindings::io::error,
+                "wasi:filesystem/preopens": crate::bindings::filesystem::preopens,
+                "wasi:sockets/network": crate::bindings::sockets::network,
+                "wasi:http": crate::bindings::http,
+
+                // Configure the resource types of the bound interfaces here
+                // to be the same as the async versions of the resources, that
+                // way everything has the same type.
+                "wasi:io/poll/pollable": super::super::io::poll::Pollable,
+                "wasi:io/streams/input-stream": super::super::io::streams::InputStream,
+                "wasi:io/streams/output-stream": super::super::io::streams::OutputStream,
+
+                // Stubbed interfaces
+                "wasi:filesystem/types/descriptor": super::super::filesystem::types::Descriptor,
+                "wasi:filesystem/types/directory-entry-stream": super::super::filesystem::types::DirectoryEntryStream,
+                "wasi:sockets/tcp/tcp-socket": crate::stubs::sockets::TcpSocket,
+                "wasi:sockets/udp/incoming-datagram-stream": crate::stubs::sockets::IncomingDatagramStream,
+                "wasi:sockets/udp/outgoing-datagram-stream": crate::stubs::sockets::OutgoingDatagramStream,
+                "wasi:sockets/udp/udp-socket": crate::stubs::sockets::UdpSocket,
+            },
+            require_store_data_send: true,
+        });
+    }
+    pub use self::generated::exports;
+    pub use self::generated::wasi::*;
+    //pub use self::generated::Command;
```

Comment:
Add note that `Command` is no longer top-level exported (previously targeting `wasi:cli/command`), since using our own root wit that includes `wasi:http` both here and in async bindings